paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -341,8 +341,20 @@ internal class PaymentSheetViewModel @Inject internal constructor(
             intentConfirmationHandler.state.collectLatest { state ->
                 when (state) {
                     is IntentConfirmationHandler.State.Idle -> Unit
+                    is IntentConfirmationHandler.State.Preconfirming -> {
+                        if (
+                            state.inPreconfirmFlow &&
+                            state.confirmationOption is PaymentConfirmationOption.GooglePay
+                        ) {
+                            setContentVisible(false)
+                        } else {
+                            setContentVisible(true)
```

Comment:
I wonder if we should pass a handle to `setContentVisible` to the intent confirmation handler, and then the actual google pay preconfirm step can be responsible for calling it, rather than doing it here. I think that would make it easier to test this behavior and also centralizes the logic for google pay "preconfirmation"

Reply:
discussed offline: we can revisit this later in the refactoring process, right now it would be a bit awkward and dealing with process death would be annoying